plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/domains/UnifiedTypeDomain.kt

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright 2010-2024 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.domains
+
+import org.jetbrains.kotlin.formver.embeddings.ClassTypeEmbedding
+import org.jetbrains.kotlin.formver.viper.MangledName
+import org.jetbrains.kotlin.formver.viper.ast.*
+
+
+const val UNIFIED_TYPE_DOMAIN_NAME = "UnifiedType"
+
+
+/**
+ * This new domain is designed to replace `NullableDomain`, `TypeDomain` and `CastingDomain` and it is not yet integrated.
+ * To enable its generation in viper output uncomment corresponding lines in
+ * [ProgramConverter](jetbrains://idea/navigate/reference?project=kotlin&path=org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt:70)
+ * and [SpecialFunctions.kt](jetbrains://idea/navigate/reference?project=kotlin&path=org/jetbrains/kotlin/formver/embeddings/callables/SpecialFunctions.kt:58)
+ *
+ * Viper code:
+ * ```viper
+ *
+ * domain UnifiedType  {
+ *
+ *
+ *  unique function intType(): UnifiedType
```

Comment:
While the domain name accurately describes what the domain + axioms do, I would use a name that points to what a value of it means.  Perhaps something like `RuntimeType`?